apisix/core/config_util.lua

```diff
@@ -56,23 +58,56 @@ end
 -- or cancelled. Note that Nginx worker exit doesn't trigger the clean handler.
 -- Return an index so that we can cancel it later.
 function _M.add_clean_handler(item, func)
-    local idx = #item.clean_handlers + 1
-    item.clean_handlers[idx] = func
-    return idx
+    if not item.clean_handlers._id then
+        item.clean_handlers._id = 1
+    end
+
+    local id = item.clean_handlers._id
+    item.clean_handlers._id = item.clean_handlers._id + 1
```

Comment:
If `item.clean_handlers._id` is nil. The result will be two because we set `item.clean_handlers._id = 1` at line 62.

Reply:
What is the `clean_handlers._id` used for?

Reply:
> What is the `clean_handlers._id` used for?

It is used to provide id for the handlers.

Reply:
> If `item.clean_handlers._id` is nil. The result will be two because we set `item.clean_handlers._id = 1` at line 62.

Yes. But we store the id (which is assigned before `item.clean_handlers._id = item.clean_handlers._id + 1`), so the first id is still 1.

Reply:
> > If `item.clean_handlers._id` is nil. The result will be two because we set `item.clean_handlers._id = 1` at line 62.
> 
> Yes. But we store the id (which is assigned before `item.clean_handlers._id = item.clean_handlers._id + 1`), so the first id is still 1.

Got it.